web-library/src/main/java/ru/skypro/lessons/springboot/weblibrary/pojo/SecurityUserDetailsService.java

```diff
@@ -0,0 +1,26 @@
+package ru.skypro.lessons.springboot.weblibrary.pojo;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+
+public class SecurityUserDetailsService implements UserDetailsService {
+    @Autowired
+    // Внедряем зависимость UserRepository для доступа к базе пользователей
+    private UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) {
+        // Ищем пользователя в базе данных с указанным именем пользователя
+        AuthUser user = userRepository.findByUsername(username);
```

Comment:
Немного редактируем метод:

```suggestion
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        // Ищем пользователя в базе данных с указанным именем пользователя
        User user = userRepository.findByUsername(username);
```